bolt-cli/src/common/web3signer.rs

```diff
@@ -0,0 +1,200 @@
+    let identity = Identity::from_pem(&identity)?;
+
+    Ok((ca_cert, identity))
+}
+
+#[derive(Serialize, Deserialize)]
+struct Keys {
+    /// The consensus keys stored in the Web3Signer.
+    pub consensus: String,
+    /// The two below proxy fields are here for deserialisation purposes.
+    /// They are not used as signing is only over the consensus type.
+    #[allow(unused)]
+    pub proxy_bls: Vec<String>,
+    #[allow(unused)]
+    pub proxy_ecdsa: Vec<String>,
+}
+
+/// Outer container for response.
+#[derive(Serialize, Deserialize)]
+struct CommitBoostKeys {
+    keys: Vec<Keys>,
+}
+
+/// Request signature from the Web3Signer.
+#[derive(Serialize, Deserialize)]
+struct CommitBoostSignatureRequest {
+    #[serde(rename = "type")]
+    pub type_: String,
+    pub pubkey: String,
+    pub object_root: String,
+}
+
+#[cfg(test)]
+pub mod test_util {
+    use std::{
+        process::{Child, Command},
+        time::Duration,
+    };
+
+    use crate::cli::Web3SignerTlsCredentials;
+    use eyre::{bail, Ok};
+
+    /// Start a Web3Signer server for testing.
+    ///
+    /// This will start a Web3Signer server and return its URL, process handle, and TLS credentials.
+    pub async fn start_web3signer_test_server(
+    ) -> eyre::Result<(String, Child, Web3SignerTlsCredentials)> {
+        let test_data_dir = env!("CARGO_MANIFEST_DIR").to_string() + "/test_data/web3signer";
+
+        // Keystore test data.
+        let keystore_dir = test_data_dir.clone() + "/keystore";
+
+        // TLS test data.
+        let tls_dir = test_data_dir.clone() + "/tls";
+        let tls_keystore = tls_dir.clone() + "/key.p12";
+        let tls_password = tls_dir.clone() + "/password.txt";
+        let ca_cert_path = tls_dir.clone() + "/web3signer.crt";
+        let combined_pem_path = tls_dir.clone() + "/combined.pem";
+
+        // Check if web3signer is installed (in $PATH).
+        if Command::new("web3signer").spawn().is_err() {
+            bail!("Web3Signer is not installed in $PATH");
+        }
+
+        // Start the web3signer server.
+        let web3signer_proc = Command::new("web3signer")
+            .arg("--key-store-path")
+            .arg(keystore_dir.clone())
+            .arg("--tls-keystore-file")
+            .arg(tls_keystore)
+            .arg("--tls-allow-any-client")
+            .arg("true")
+            .arg("--tls-keystore-password-file")
+            .arg(tls_password.clone())
+            .arg("eth2")
+            .arg("--network")
+            .arg("mainnet")
+            .arg("--slashing-protection-enabled")
+            .arg("false")
+            .arg("--commit-boost-api-enabled")
+            .arg("true")
+            .arg("--proxy-keystores-path")
+            .arg(keystore_dir.clone())
+            .arg("--proxy-keystores-password-file")
+            .arg(tls_password)
+            .spawn()?;
+
+        // Allow the server to start up.
+        tokio::time::sleep(Duration::from_secs(5)).await;
```

Comment:
IIRC this code would not output to stdout. Can we add a little log so that the user knows why it is waiting?

Reply:
spawn() does log to stdout during the test, so this in particular is not needed in my opinion